library/public/window.h

```diff
@@ -6,16 +14,45 @@
 
 #include <string>
 
-// TODO DOC
 namespace f3d
 {
 class F3D_EXPORT window
 {
 public:
+  /**
+   * Use all the rendering related options to update the configuration of the window
+   * and the rendering stack below. This also initialize the rendering stack if needed.
+   * This will be called automatically when calling loader::loadFile but can also be called manually
+   * when needed. This must be called, either manually or automatically, before any render call.
+   * Return true on success, false otherwise.
```

Comment:
I think we should be consistent and more modern on our API. We should throw in case of failure, and we should return `window&` to chain calls, what do you think?  
Maybe not for `update()`, but doing `window.setIcon().setWindowName()` make sense to me.

Reply:
Lets keep as it is now but discuss the API in details in #52